src/main/java/it/gov/pagopa/payhub/auth/service/a2a/AuthorizeClientCredentialsRequestService.java

```diff
@@ -3,24 +3,54 @@
 import it.gov.pagopa.payhub.auth.exception.custom.ClientUnauthorizedException;
 import it.gov.pagopa.payhub.auth.mapper.ClientMapper;
 import it.gov.pagopa.payhub.model.generated.ClientDTO;
+import org.springframework.beans.factory.annotation.Value;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
 
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 @Service
 @Slf4j
 public class AuthorizeClientCredentialsRequestService {
+	private static final String ERROR = "Unauthorized client for client-credentials";
+	private static final String REGEX = "^(\\w+\\s*)(piattaforma-unitaria\\b)$";
```

Comment:
The clientId will be "piattaforma-unitaria_"+IPACODE
let's use startsWith, not a regexp, it will perform better